src/main/java/dtu/scheduler/SchedulingApp.java

```diff
@@ -43,13 +47,32 @@ public void assignActivity(String workerId, Activity activity) throws WorkerDoes
 		activityAssigner.assignActivity(getWorkerById(workerId), activity);
 	}
 	
-
-	public SchedulingApp() {
-	}
-
 	public double getWeeklyRegisteredHours() {
 		return currentUser.getWeeklyRegisteredHours();
 	}
+	
+	public void addProject(Project project) throws ProjectAlreadyExistException {
+		// Test if project already exists
+		if (searchProject(project.getProjectID()) != null) {
+			throw new ProjectAlreadyExistException("Project already exist");
+		}
+		
+		projectArray.add(project);
+	}
+	
+	public Project searchProject(String ID) {
+		for (int i = 0; i < projectArray.size(); i++) {
+			Project p = projectArray.get(i);
+			if (p.getProjectID().equals(ID)) {
+				return p;
```

Comment:
Refactor according to Single Responsibility priciple. redelegate work to new class